src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -373,7 +373,7 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
         //
         // Code below this check won't lag each tick anymore
         //
-        if (getColony() == null || !getColony().shallUpdate(event.world, ONWORLD_TICK_AVERAGE))
+        if (!Colony.shallUpdate(event.world, ONWORLD_TICK_AVERAGE))
```

Comment:
Revert. Using methods is always better then fields!

Reply:
calling a static method on a field?

Reply:
-- solved confusion, shallUpdate is static method, so it should be called via static refence --

Reply:
ooh, i missed that `shallUpdate()` is `static` so it should be called this way, of course
and sorry guys about `event.world`, was too tired after reading lots of minecolonies code, shit happens :(
looks like i've lost my skills, time to get them back...

Reply:
No problem, I found it fast =D

Reply:
Keep contributing and you'll get into it, but I advise you to install Intellij because you can run Minecraft with it to debug if your fixes are really working.